src/docmanager/widgetmanager.ts

```diff
@@ -69,29 +65,32 @@ class DocumentWidgetManager implements IDisposable {
   }
 
   /**
-   * Test whether the context has been disposed (read-only).
+   * Test whether the document widget manager is disposed.
    */
   get isDisposed(): boolean {
     return this._registry === null;
   }
 
   /**
-   * Dispose of the resources used by the widget manager.
+   * Dispose of the resources used by the widget.
    */
   dispose(): void {
     if (this.isDisposed) {
       return;
     }
+    disconnectReceiver(this);
```

Comment:
We don't define any of our own signals, but we connect to other signals, so this is more explicit.